src/GraphQL.Primitives/GraphQLQuery.cs

```diff
@@ -1,15 +1,34 @@
-#if NET6_0_OR_GREATER
 using System.Diagnostics.CodeAnalysis;
-
 namespace GraphQL;
 
 /// <summary>
-/// Value record for a GraphQL query string
+/// Value object representing a GraphQL query string and storing the corresponding APQ hash. <br />
+/// Use this to hold query strings you want to use more than once.
 /// </summary>
-/// <param name="Text">the actual query string</param>
-public readonly record struct GraphQLQuery([StringSyntax("GraphQL")] string Text)
+public class GraphQLQuery : IEquatable<GraphQLQuery>
```

Comment:
Here's some different thoughts while looking at the changes here:
- Why moving from a struct to a class?
- Why not lazy load the hash?  Maybe more important to preload the hash if it was a struct, but we are using a class here.
- String comparisons will automatically short-circuit if they point to the same reference, so the benefit of comparing the Sha256Hash vs the Text is likely very minimal.

Not sure I have a concrete suggestion, but those are my thoughts.

Reply:
I had introduced the GraphQLQuery struct mainly to provide support the [StringSyntax("GraphQL")] attribute when declaring query strings as variables (so just a wrapper around the query string).

> Why moving from a struct to a class?

I went from struct to class because the reference type characteristics are (at least in my understanding) especially benefical for the APQ concept (i.e. declare ALL queries your application uses in one GraphQLQuery, send that once and then leverage APQ and operation name), thus reusing the same reference everywhere.

> Why not lazy load the hash?

I think it makes sense for the intended use of the GraphQLQuery class to compute the hash on object construction. It's propably something you rather want to happen during application startup than later when you are waiting for a query.
If you don't want to pre-compute the hash, just don't use the GraphQLQuery class and go with strings.

Reply:
Makes sense